docs/executor_tutorial/google_lifesciences.rst

```diff
@@ -0,0 +1,880 @@
+results (under sorted_reads and calls) are saved there too!
+
+.. image:: workflow/results-google-storage.png
+
+The source folder contains a cache folder with archives that contain your working directories,
+one for each step. You can safely delete this folder, or keep it if you want to reproduce
+the run in the future.
+
+
+Step 5: Debugging
+:::::::::::::::::
+
+Let's introduce an error (purposefully) into our Snakefile to practice debugging.
+Let's remove the conda environment.yml file for the first rule, so we would
+expect that Snakemake won't be able to find the executables for bwa and samtools.
+In your Snakefile, change this:
+
+.. code:: python
+
+    rule bwa_map:
+        input:
+        fastq="samples/{sample}.fastq",
+        idx=multiext("genome.fa", ".amb", ".ann", ".bwt", ".pac", ".sa")
+    conda:
+        "environment.yml"
+    output:
+        "mapped_reads/{sample}.bam"
+    params:
+        idx=lambda w, input: os.path.splitext(input.idx[0])[0]
+    shell:
+        "bwa mem {params.idx} {input.fastq} | samtools view -Sb - > {output}"
+
+
+to this:
+
+.. code:: python
+
+    rule bwa_map:
+        input:
+        fastq="samples/{sample}.fastq",
+        idx=multiext("genome.fa", ".amb", ".ann", ".bwt", ".pac", ".sa")
+    output:
+        "mapped_reads/{sample}.bam"
+    params:
+        idx=lambda w, input: os.path.splitext(input.idx[0])[0]
+    shell:
+        "bwa mem {params.idx} {input.fastq} | samtools view -Sb - > {output}"
+
+
+And then you'll need to remove the plots, mapped_reads, and calls folders
+so the step will run again. Then, run the job again:
+
+.. code:: console
+
+    snakemake --google-lifesciences --default-remote-prefix snakemake-testing-data --use-conda --google-lifesciences-region us-west1 --container-image snakemake/snakemake:v5.10.0
+
+
+Everything will start out okay as it did before, and it will pause on the first 
+step when it's deploying the first container image. The last part of the 
+log will look somethig like this:
+
+
+.. code:: console
+
+    [Fri Apr 17 22:01:38 2020]
+    rule bwa_map:
+        input: snakemake-testing-data/samples/B.fastq, snakemake-testing-data/genome.fa.amb, snakemake-testing-data/genome.fa.ann, snakemake-testing-data/genome.fa.bwt, snakemake-testing-data/genome.fa.pac, snakemake-testing-data/genome.fa.sa
+        output: snakemake-testing-data/mapped_reads/B.bam
+        jobid: 8
+        wildcards: sample=B
+        resources: mem_mb=15360, disk_mb=128000
+
+    Get status with:
+    gcloud config set project snakemake-testing
+    gcloud beta lifesciences operations describe projects/snakemake-testing/locations/us/operations/11698975339184312706
+    gcloud beta lifesciences operations list
+
+
+If all went okay, you'd probably ignore this. But since we removed an important
+dependency to install libraries with conda, we are definitely going to hit
+an error! That looks like this:
```

Comment:
So, ideally, the rule has a log file, in which case we don't see this error in the stderr but instead we can look into the logfile. After this kind of debugging is discussed, please add a section below that explains that adding a log file to the rule is preferable, and how to look for errors in there.

Reply:
Good idea! I'll add this.